### DISCUSSION TITLE
openssl setup problem

### DISCUSSION THREAD

**Discussion by danmillwood** [zopencommunity/meta#332] [Q&A]
Hello, I had my first attempt at installing open tools today.  I installed `openssl` via `curl` which worked

```
Setting up openssl...
Setup completed.
```

but when I try to use it, its looking in the wrong place for some of the files.  I installed into `/u/myname/opentools/openssl-1.1.1t` but when I run the command I get the following:

```
$ openssl req -newkey rsa:2048 -keyout PRIVATEKEY.key -out MYCSR.csr                                       
Can't open /u/other/zopen/prod/openssl-1.1.1t/ssl/openssl.cnf for reading, EDC5129I No such file or directory.
2367961018136526849:error:02001081:system library:fopen:reason(129):crypto/bio/bss_file.c:69:fopen('/u/other/zopen/prod/openssl-1.1.1t/ssl/openssl.cnf','r')
2367961018136526849:error:2006D080:BIO routines:BIO_new_file:no such file:crypto/bio/bss_file.c:76:
```

Im trying to figure out where it gets the directory name `/u/other/zopen` from.  `/u/other` exists, but the directory `zopen` does not.  Any help appreciated. It's not in any of my environment variables.  Thanks

**Comment by MikeFultonDev** [comment]
Hm. Is it really ‘other’ or do you just mean another id? 
is anything from /u/other in your PATH that might be confusing the install? @IgorTodorovskiIBM ideas?

**Reply by IgorTodorovskiIBM** [reply]
Hmm, any symlinks or environment variables that point to /u/other?  
Is `OPENSSL_CONF` environment variable set? What if you set OPENSSL_CONF to the correct location, does it override the default?

**Reply by danmillwood** [reply]
It wasn't actually `/u/other`, I was just avoiding sharing real userids.  Under `/u/other` I do have a set of rocket open source tools, but the new tools are first in the PATH, LIBPATH etc in my profile.  

I didn't have `OPENSSL_CONF` set, only `OPENSSL_HOME=/u/myname/opentools/openssl-1.1.1t` after sourcing the `.env` file.  I added `export OPENSSL_CONF=/u/myname/opentools/openssl-1.1.1t/ssl/openssl.cnf` and it started working.  Thanks!

**Comment by MikeFultonDev** [comment]
@danmillwood I know this was awhile ago - we have a new drop of 'zopen' that hopefully is much 'cleaner'. If you can try it, would love to get your feedback. See: https://zosopentools.github.io/meta/#/Guides/QuickStart to get started.
